docs/en/resource/supplemental.md

```diff
@@ -0,0 +1,351 @@
+// was passed, a type error would occur
+```
+
+## API
+
+The data middleware API provides the widget with access to the resource data.
+
+### get options
+
+`getOptions` returns the current `Options` object. The result of this function should be passed to each of the `get` api functions. It is important to use `getOptions` rather than constructing a new `Options` object in order to ensure that shared resources etc work as expected.
+
+### set options
+
+`setOptions` is used to update the current `Options` on the resource wrapper. This will reflect in any other widget using the shared resource.
+
+```ts
+const { getOptions, setOptions } = data();
+
+setOptions({
+	...getOptions(),
+	pageNumber: 1
+});
+```
+
+### get
+
+The `get` function takes an `Options` object and returns data as requested if it is already available on the resource. It will not perform a `read`.
+
+```ts
+const { getOptions, get } = data();
+
+const data = get(getOptions());
+```
+
+### getorread
+
+The `getOrRead` function takes an `Options` object and returns the data requested. If it is not already availablee, it will perform a `read`. When the data is available the widget will be invalidated so it can be used in a reactive manner.
+
+```ts
+const { getOptions, getOrRead } = data();
+
+const data = getOrRead(getOptions());
+```
+
+### gettotal
+
+The `getTotal` function takes an `Options` object and returns the current total for the given `query`.
+When the total changes or becomes available, the widget will be invalidated.
+
+```ts
+const { getOptions, getTotal } = data();
+
+const total = getTotal(getOptions());
+```
+
+### isLoading
+
+The `isLoading` function takes an `Options` object and returns a `boolean` to indicate if there is a in flight read in progress for the current `Options`.
+
+```ts
+const { getOptions, isLoading } = data();
+
+const loading = isLoading(getOptions());
+```
+
+### isFailed
+
+The `isFailed` function takes an `Options` object and returns a `boolean` to indicate if a read with the current `Options` has failed.
+
+```ts
+const { getOptions, isFailed } = data();
+
+const failed = isFailed(getOptions());
+```
+
+## Data Initialiser Options
+
+### Pass in resource
+
+Where the `resource` is passed via a different property or there are multiple resources passed, you can pass the resource directly to the `data` initialiser.
+
+```ts
+const { namedResource } = properties();
+const { getOptions } = data({ resource: namedResource });
+```
+
+### Reset resource
+
+Where a widget does not wish to share a resource and wants to ensure it always gets it's own set of `Options`, a reset initialiser is available. This may be appropriate when creating a typeahead or other such filtering component. You may not want to risk changing the data query used elsewhere in the application to the `reset` option should be used. This option even creates a new `Options` object when your widget has been passed a `shared` resource.
```

Comment:
```suggestion
Widgets that do not want to share a resource and need to ensure their resource always gets its own set of `Options` can indicate so with the `reset` initializer. This may be appropriate when creating a typeahead or other such filtering component, where a data query may be shared between multiple locations within an application. Each widget instance should ideally require its own unique copy to avoid inadvertently changing data elsewhere in the application. This option will also create a new `Options` object for widgets that have been passed a `shared` resource.
```